src/NServiceBus.Testing/NSB.Testing.Fakes/TestingLoggerFactory.cs

```diff
@@ -14,39 +15,72 @@ public class TestingLoggerFactory : LoggingFactoryDefinition
         /// </summary>
         public TestingLoggerFactory()
         {
-            level = new Lazy<LogLevel>(() => LogLevel.Debug);
-            writer = new Lazy<TextWriter>(() => TextWriter.Null);
+            lazyLevel = new Lazy<LogLevel>(() => LogLevel.Debug);
+            lazyWriter = new Lazy<TextWriter>(() => TextWriter.Null);
         }
 
         /// <summary>
-        /// Controls the <see cref="Logging.LogLevel" />.
+        /// Controls the <see cref="Logging.LogLevel" /> for all default logging.
         /// </summary>
+        /// <param name="level">The log level to be used.</param>
         public void Level(LogLevel level)
         {
-            this.level = new Lazy<LogLevel>(() => level);
+            lazyLevel = new Lazy<LogLevel>(() => level);
         }
 
         /// <summary>
-        /// Instructs the logger to write to the provided text writer.
+        /// Instructs the logger to write to the provided text writer for all default logging.
         /// </summary>
         /// <param name="writer">The text writer to be used.</param>
         public void WriteTo(TextWriter writer)
         {
-            this.writer = new Lazy<TextWriter>(() => writer);
+            lazyWriter = new Lazy<TextWriter>(() => writer);
+        }
+
+        /// <summary>
+        /// Instructs the logger to write to the provided text writer for the given scope.
+        /// </summary>
+        /// <param name="writer">The text writer to be used.</param>
+        /// <param name="level">The log level to be used.</param>
+        /// <returns>The logging scope. Cannot be nested.</returns>
+        public IDisposable Use(TextWriter writer, LogLevel level = LogLevel.Debug)
```

Comment:
I think this being called `Use` is a bit confusing given the factory setup also uses the same name and you'd be chaining `Use`. Maybe `CreateScope` or something like that?

Reply:
Good suggestion. Haven't thought about it from that angle. How about

```
IDisposable Scope(TextWriter writer, LogLevel level = LogLevel.Debug)
```

or

```
IDisposable BeginScope(TextWriter writer, LogLevel level = LogLevel.Debug)
```

or

```
IDisposable CreateScope(TextWriter writer, LogLevel level = LogLevel.Debug)
```

which one do you prefer?

Reply:
given that you don't have to explicitly use the created scope but it's being applied automatically, I'm leaning towards `BeginScope`

Reply:
Renamed to `BeginScope`